File: softeer/HSAT_7_자동차테스트.java
import java.io.*;
import java.util.*;

// 통과 1.230초 52.04MB 
public class Main {

    static int fuel[], medians[], n, q;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");

        n = Integer.parseInt(st.nextToken()); // 자동차 수
        q = Integer.parseInt(st.nextToken()); // 테스트 수
        fuel = new int[n]; // 자동차별 연비
        medians = new int[q]; // 테스트 별 중앙값

        st = new StringTokenizer(br.readLine(), " ");
        for(int i = 0; i < n; i++) {
            fuel[i] = Integer.parseInt(st.nextToken());
        }

        for(int i = 0; i < q; i++) {
            medians[i] = Integer.parseInt(br.readLine());
        }

        Arrays.sort(fuel);
        for(int m : medians) {
            int ans = 0;
            // small, m, big 의 형태로 나오는 경우의 수는 small 경우 수 * big 경우 수
            int mediIdx = binarySearch(0, n-1, m); // 중앙값 인덱스 구하기
            if(mediIdx > 0) {
                ans = mediIdx * (n - mediIdx - 1);
            }
            System.out.println(ans);
        }
        
    }

    private static int binarySearch(int left, int right, int target) {
        int mid = 0;
        while(left < right) {
            mid = (left + right) / 2;
            if(fuel[mid] == target) {
                return mid;
            } else if(fuel[mid] < target) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }
        return -1;
    }
}

File: softeer/HSAT_7_자동차테스트.js
const input = require('fs').readFileSync('/dev/stdin', 'utf8').replace(/\n+/g, '\n').trim().split('\n');
const num = input[0].split(' ').map(Number);
const n = num[0];
const q = num[1];
const fuel = input[1].trim().split(' ').map(Number).sort((a, b) => a - b); // 그냥 sort()만 하면 문자열로 정렬됨.
const medians = input.slice(2).map(Number);

// 통과 0.268초 72.86MB 메모리
let res = [];
for(let m of medians) {
    let ans = 0;
    let medIdx = binarySearch(0, n - 1, m);
    if(medIdx > 0) {
        ans = medIdx * (n - medIdx - 1);
    }
    res.push(ans);
}
console.log(res.join('\n'));

function binarySearch(left, right, target) {
    let mid = 0;
    while(left < right) {
        mid = Math.floor((left + right) / 2);
        if(fuel[mid] == target) {
            return mid;
        } else if(fuel[mid] < target) {
            left = mid + 1;
        } else {
            right = mid;
        }
    }
    return -1;
}
